File: java/lib/src/main/java/adf_core_python/core/gateway/message/MAModuleResponse.java
package adf_core_python.core.gateway.message;

import adf_core_python.core.gateway.message.urn.ModuleMessageComponentURN;
import adf_core_python.core.gateway.message.urn.ModuleMessageURN;
import rescuecore2.messages.AbstractMessage;
import rescuecore2.messages.components.StringComponent;
import rescuecore2.messages.protobuf.RCRSProto;

import java.io.IOException;
import java.io.InputStream;

public class MAModuleResponse extends AbstractMessage {
    private final StringComponent moduleID;
    private final StringComponent className;

    public MAModuleResponse(String moduleID, String className) {
        this();
        this.moduleID.setValue(moduleID);
        this.className.setValue(className);
    }

    public MAModuleResponse(InputStream inputStream) throws IOException {
        this();
        read(inputStream);
    }

    public MAModuleResponse(RCRSProto.MessageProto messageProto) {
        this();
        fromMessageProto(messageProto);
    }

    private MAModuleResponse() {
        super(ModuleMessageURN.MA_MODULE_RESPONSE);
        this.moduleID = new StringComponent(ModuleMessageComponentURN.ModuleID);
        this.className = new StringComponent(ModuleMessageComponentURN.ClassName);
        addMessageComponent(moduleID);
        addMessageComponent(className);
    }

    public String getModuleID() {
        return this.moduleID.getValue();
    }

    public String getClassName() {
        return this.className.getValue();
    }
}

File: src/adf_core_python/core/gateway/message/ma_module_response.py
from abc import ABC
from typing import Optional

from rcrscore.messages import KAControlMessage
from rcrscore.proto import RCRSProto_pb2
from rcrscore.urn.control_message import ControlMessageURN

from adf_core_python.core.gateway.message.urn.urn import (
  ComponentModuleMSG,
  ModuleMSG,
)


class MAModuleResponse(KAControlMessage, ABC):
  def __init__(self, message_proto: RCRSProto_pb2.MessageProto) -> None:
    self.module_id: Optional[str] = None
    self.class_name: Optional[str] = None
    self.read(message_proto)

  def read(self, message_proto: RCRSProto_pb2.MessageProto) -> None:
    self.module_id = message_proto.components[ComponentModuleMSG.ModuleID].stringValue
    self.class_name = message_proto.components[ComponentModuleMSG.ClassName].stringValue

  @staticmethod
  def get_urn() -> ControlMessageURN:
    return ModuleMSG.MA_MODULE_RESPONSE  # type: ignore
